Mathlib/Algebra/Order/Group/Synonym.lean

```diff
@@ -161,6 +161,15 @@ instance [h : Semigroup α] : Semigroup (Lex α) := h
 @[to_additive]
 instance [h : CommSemigroup α] : CommSemigroup (Lex α) := h
 
+@[to_additive]
+instance [Mul α] [h : IsLeftCancelMul α] : IsLeftCancelMul (Lex α) := h
```

Comment:
Can these just use `inferInstanceAs`? Named instances are non-idiomatic.

Reply:
There is a ton of unchanged code above this which has this format.

Reply:
I agree but we shouldn't introduce new instances unless they are absolutely necessary. Unsure if this should hold things up but seems like an easy fix. 
```suggestion
instance [Mul α] [IsLeftCancelMul α] : IsLeftCancelMul (Lex α) := inferInstanceAs <| IsLeftCancelMul α
```
Haven't checked this works.